storage/aptosdb/src/state_store/mod.rs

```diff
@@ -1169,18 +1172,40 @@ impl StateValueWriter<StateKey, StateValue> for StateStore {
             self.state_kv_db.enabled_sharding(),
         )?;
         self.state_kv_db
-            .commit(version, batch, sharded_schema_batch)
+            .commit(version, batch, sharded_schema_batch)?;
+        if self.internal_indexer_db.is_some() {
```

Comment:
probably want to write index first since you are returning the main db progress in `get_progress`?